client/Client.cpp

```diff
@@ -593,7 +598,7 @@ void CClient::battleStarted(const BattleInfo * info)
 
 	if(info->tacticDistance && vstd::contains(battleints, info->sides[info->tacticsSide].color))
 	{
-		boost::thread(&CClient::commenceTacticPhaseForInt, this, battleints[info->sides[info->tacticsSide].color]);
+		playerTacticThreads[info->sides[info->tacticsSide].color] = std::make_unique<boost::thread>(boost::thread(&CClient::commenceTacticPhaseForInt, this, battleints[info->sides[info->tacticsSide].color]));

```

Comment:
does it make sense to check if playerTacticThreads[info->sides[info->tacticsSide].color] thread is finished before creating a new thread?

Reply:
How can I check this? Call join() if the thread exists?